### PR TITLE
Add checkboxes to signup and verification process

### PR DESCRIPTION
**Pull request by beagleknight**
# What and why

I added a checkbox checked by default when a user is in the sign up form to receive newsletters. The same checkbox can be unchecked from user's account.

Also on the address verification process I added a dynamic checkbox which appears when the user is 16 or 17 years old.
# QA

Sign up with a new user. The newsletter checkbox must be checked by default. Then when you try to verify the account if you select you date of birth to have 16 or 17 years another mandatory checkbox should appear.
# GIF Tax

### CODE REVIEW

File: app/models/verification/residence.rb [hunk 1]
```diff
@@ -65,6 +66,10 @@ def store_failed_attempt
     })
   end
 
+  def adult?
+    (Date.today.year - self.date_of_birth.year) > 18
```
Comment (josepjaume) [inline app/models/verification/residence.rb]: Careful with this as one could be marked as an adult before (imagine someone that was born in the 31st December/ 1998).

Comment (beagleknight) [inline app/models/verification/residence.rb]: hmmm I see your point, let me fix it. So.. I should check the javascript part as well.

Comment (josepjaume) [inline app/models/verification/residence.rb]: I don't think you need to check the JS part - just get the difference of dates _first_ and then check if it's more or less than 18 years :dancer:

Comment (josepjaume) [inline app/models/verification/residence.rb]: Wait this is not trivial xD.

Comment (josepjaume) [inline app/models/verification/residence.rb]: https://github.com/tmlee/time_difference

**Comment by josepjaume** [comment]
:+1:

**Pull request merged**